client/blocks/announcement-modal/docs/example.jsx

```diff
@@ -1,20 +1,55 @@
+import { useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PlaceholderImage from 'calypso/assets/images/marketplace/plugins-revamp.png';
 import AnnouncementModal from 'calypso/blocks/announcement-modal';
-import PlaceholderImage from 'calypso/images/extensions/woocommerce/image-placeholder.png';
+import Button from 'calypso/components/forms/form-button';
+import { getCurrentUserId } from 'calypso/state/current-user/selectors';
+import { savePreference } from 'calypso/state/preferences/actions';
+import { hasReceivedRemotePreferences } from 'calypso/state/preferences/selectors';
+
+const AnnouncementModalExample = () => {
+	const announcementId = 'announcement-modal-example';
+	const [ show, setShow ] = useState( false );
+	const dispatch = useDispatch();
+	const userId = useSelector( ( state ) => getCurrentUserId( state ) );
+	const hasPreferences = useSelector( hasReceivedRemotePreferences );
+	const dismissPreference = `announcement-modal-${ announcementId }-${ userId }`;
```

Comment:
Just wanted to confirm we're intentionally tinkering with real user preferences here.

Reply:
Yes, that is intentional. I would supper happy to change this though, is there another workaround for a component that hides itself depending on user preferences?
https://github.com/Automattic/wp-calypso/blob/0178a4ae1aa2031071dc846cfc0344b353fd233e/client/blocks/announcement-modal/index.jsx#L46-L48

Reply:
Guess it could be a good idea to allow for clearing / resetting the user preference from within the example, but that's not a requirement I guess.